iceoryx_posh/test/moduletests/test_roudi_portmanager.cpp

```diff
@@ -562,3 +571,133 @@ TEST_F(PortManager_test, PortDestroy)
         }
     }
 }
+
+TEST_F(PortManager_test, CheckNodeOverflowErrorReturnsnullptrForAcquireNodeData)
+{
+    std::string nodename = "nodeName";
+
+    for (uint32_t i = 0U; i < iox::MAX_NODE_NUMBER; i++)
+    {
+        auto newNodeName = nodename + std::to_string(i);
+        auto nodePointer = m_portManager->acquireNodeData(iox::ProcessName_t(iox::cxx::TruncateToCapacity, newNodeName),
+                                                          iox::NodeName_t(iox::cxx::TruncateToCapacity, newNodeName));
+        EXPECT_NE(nodePointer, nullptr);
+    }
+
+    { // test if overflow errors get hit
+
+        auto errorHandlerCalled{false};
+        auto errorHandlerGuard = iox::ErrorHandler::SetTemporaryErrorHandler(
+            [&errorHandlerCalled](const iox::Error, const std::function<void()>, const iox::ErrorLevel) {
+                errorHandlerCalled = true;
+            });
+
+        auto nodePointer = m_portManager->acquireNodeData(m_Process, m_Node);
+        EXPECT_EQ(nodePointer, nullptr);
+        EXPECT_TRUE(errorHandlerCalled);
+    }
+
+    // delete one and add one should be possible now
+    {
+        unsigned int testi = 0;
+        auto newNodeName = nodename + std::to_string(testi);
+        m_portManager->deletePortsOfProcess(iox::ProcessName_t(iox::cxx::TruncateToCapacity, newNodeName));
+
+        auto nodePointer = m_portManager->acquireNodeData(iox::ProcessName_t(iox::cxx::TruncateToCapacity, newNodeName),
+                                                          iox::NodeName_t(iox::cxx::TruncateToCapacity, newNodeName));
+        EXPECT_NE(nodePointer, nullptr);
+    }
+}
+
+TEST_F(PortManager_test, UseDestroyNodeReturnsNotNullPtrToAcquireNodeData)
+{
+    std::vector<iox::runtime::NodeData*> nodeContainer;
+    std::string nodeName = "nodeName";
+
+    // first aquire all possible condition variables
+    for (uint32_t i = 0U; i < iox::MAX_NODE_NUMBER; i++)
+    {
+        auto newnodeName = nodeName + std::to_string(i);
+        auto nodeDataResult =
+            m_portManager->acquireNodeData(m_Process, iox::NodeName_t(iox::cxx::TruncateToCapacity, newnodeName));
+        EXPECT_NE(nodeDataResult, nullptr);
+        nodeContainer.push_back(nodeDataResult);
+    }
+
+    // so now no one should be available
+    {
+        auto nodeDataResult = m_portManager->acquireNodeData(m_Process, m_Node);
+        EXPECT_EQ(nodeDataResult, nullptr);
+    }
+
+    // set the destroy flag and let the discovery loop take care
+    for (uint32_t i = 0U; i < iox::MAX_NODE_NUMBER; i++)
+    {
+        nodeContainer[i]->m_toBeDestroyed.store(true, std::memory_order_relaxed);
```

Comment:
@elBoberido is it the same case in this two old tests in the same file:
DestroyNodeDataAndAddNewNodeDataSucceeds and ConditionVariablesDestroy

Reply:
@manel94 yes, they have the same issue. A bad implementation can potentially crash the test. They should also have an `ASSERT_*` instead of an `EXPECT_*` in order to immediately stop the test case and prevent a nullptr access later on.

Reply:
Okay i will update it also as a refactoring